core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -328,16 +328,15 @@ private[spark] object TaskMetrics extends Logging {
    */
   def fromAccumulators(accums: Seq[AccumulatorV2[_, _]]): TaskMetrics = {
     val tm = new TaskMetrics
-    for (acc <- accums) {
+    val (innerAccums, externalAccums) = accums.
```

Comment:
Should we extract a local variable `val nameToAccums = tm.nameToAccums`? `tm.nameToAccums` is lazy and frequent access may have some cost.

Reply:
Hmm, I does not understand what you mean？ tm.nameToAccums is a fixed size
 LinkedHashMap, why do we need create another local variable？Did I miss something else?

Reply:
> Should we extract a local variable `val nameToAccums = tm.nameToAccums`? `tm.nameToAccums` is lazy and frequent access may have some cost.

Hmm, I does not understand what you mean？ tm.nameToAccums is a fixed size
 LinkedHashMap, why do we need create another local variable？Did I miss something else?

Reply:
> I think the concern was that each call to `.nameToAccums` under the hood goes through a `lazy val` which is slightly more expensive to access than an ordinary field.
> 

i gotten.